### PR TITLE
Split file system watching integration tests

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesBetweenBuildsFileSystemWatchingIntegrationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        when:
+        buildKotlinFile.text = """
+            println("Hello from the modified build!")
+        """
+        withWatchFs().run "help"
+        then:
+        outputContains "Hello from the modified build!"
+    }
+
+    def "settings script changes get recognized"() {
```
Comment (lptr) [inline subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesBetweenBuildsFileSystemWatchingIntegrationTest.groovy:108]: Nit: "are" instead of "get"?

File: subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesByGradleFileWatchingIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        when:
+        localStateOutputFile.delete()
+        waitForChangesToBePickedUp()
+        withWatchFs().withBuildCache().run("localStateTask", "outputFileTask")
+        then:
+        skipped(":localStateTask")
+        executedAndNotSkipped(":outputFileTask")
+    }
+
+    // This makes sure the next Gradle run starts with a clean BuildOutputCleanupRegistry
```
Comment (lptr) [inline subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesByGradleFileWatchingIntegrationTest.groovy:254]: Why is this required?

Comment (wolfs) [reply subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesByGradleFileWatchingIntegrationTest.groovy:254]: It is used by `detects when stale outputs are removed` to trigger stale output cleanup.

File: subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesBetweenBuildsFileSystemWatchingIntegrationTest.groovy (hunk 1 above)
Comment (wolfs) [reply subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesBetweenBuildsFileSystemWatchingIntegrationTest.groovy]: Fixed.
